File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/socket/entity/request/SendFeedbackRequest.java
package ru.usedesk.chat_sdk.internal.data.framework.socket.entity.request;

import com.google.gson.annotations.SerializedName;

import ru.usedesk.chat_sdk.external.entity.UsedeskFeedback;

public class SendFeedbackRequest extends BaseRequest {
    private static final String TYPE = "@@server/chat/CALLBACK";

    private static final String KEY_DATA = "data";
    private static final String VALUE_FEEDBACK_ACTION = "action";
    private final Payload payload;

    public SendFeedbackRequest(String token, UsedeskFeedback feedback) {
        super(TYPE, token);
        payload = new Payload(feedback);
    }

    private class Payload {

        private final String type;

        @SerializedName(KEY_DATA)
        private final UsedeskFeedback feedback;

        Payload(UsedeskFeedback feedback) {
            type = VALUE_FEEDBACK_ACTION;
            this.feedback = feedback;
        }
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/socket/entity/request/SendFeedbackRequest.kt
package ru.usedesk.chat_sdk.internal.data.framework.socket.entity.request

import com.google.gson.annotations.SerializedName
import ru.usedesk.chat_sdk.external.entity.UsedeskFeedback

class SendFeedbackRequest(
        token: String,
        feedback: UsedeskFeedback
) : BaseRequest(TYPE, token) {

    private val payload: Payload

    private class Payload internal constructor(feedback: UsedeskFeedback) {
        private val type: String = VALUE_FEEDBACK_ACTION

        @SerializedName(KEY_DATA)
        private val feedback: UsedeskFeedback = feedback
    }

    companion object {
        private const val TYPE = "@@server/chat/CALLBACK"
        private const val KEY_DATA = "data"
        private const val VALUE_FEEDBACK_ACTION = "action"
    }

    init {
        payload = Payload(feedback)
    }
}